lib/tether/ops_linux.go

```diff
@@ -661,13 +663,33 @@ func (t *BaseOperations) MountLabel(ctx context.Context, label, target string) e
 	}
 
 	if err := Sys.Syscall.Mount(label, target, "ext4", syscall.MS_NOATIME, ""); err != nil {
+		// consistent with MountFileSystem
 		detail := fmt.Sprintf("mounting %s on %s failed: %s", label, target, err)
 		return errors.New(detail)
 	}
 
 	return nil
 }
 
+// MountTarget performs a mount based on the target path from the source url
+// This assumes that the source url is valid and available.
+func (t *BaseOperations) MountTarget(ctx context.Context, source url.URL, target string, mountOptions string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("Mounting %s on %s", source.String(), target)))
+
+	if err := os.MkdirAll(target, 0644); err != nil {
+		// same as MountLabel error for consistency
+		return fmt.Errorf("unable to create mount point %s: %s", target, err)
+	}
+
+	rawSource := source.Hostname() + ":/" + source.Path
```

Comment:
I usually avoid using `SprintF` since it is an order of magnitude slower thank concatentation. Though it might be an order of magnitude greater in readability. I may not make this change this time around. The Tether is going to be rewritten at some point right?